khmer/kfile.py

```diff
@@ -167,3 +169,45 @@ def check_valid_file_exists(in_files):
         else:
             print('WARNING: Input file %s not found' %
                   in_file, file=sys.stderr)
+
+
+def is_block(fthing):
+    """Takes in a file object and checks to see if it's a block or fifo"""
+
+    if fthing is sys.stdout or fthing is sys.stdin:
+        return True
+    else:
+        mode = os.stat(fthing.name).st_mode
+        return S_ISBLK(mode) or S_ISCHR(mode)
```

Comment:
I would assume if I was doing a file size check that I could use `is_block` to skip some files.